### ISSUE TITLE
Can the first argument in the mapSources function be the full, absolute file path?

### ISSUE THREAD

**Issue by azaslavsky** [floridoo/gulp-sourcemaps#274]
Currently, when I do this:

```javascript
return gulp.src(files)
      .pipe(prettyError())
      .pipe(sourcemaps.init())
      /* plugins etc */
      .pipe(sourcemaps.write('./', {
        mapSources: function(sourcePath, index) {
          console.log(sourcePath);
          return files[index].replace(process.cwd(), '');
        }
      }))
```

The logged "sourcePath" names end up being only the file name rather than the absolute file path (`thing.js` instead of `/Users/whatever/myProject/src/thing.js`).  The latter seems much more useful, especially in my case. where I have several different files in different directories all sharing the same name (`models/thing.js`, `views/thing.js`, etc).

**Comment by nmccready** [comment]
Could pass the source, then the file, or the absolute path. But the  original filePath that is in sources would have to be first.

**Comment by nmccready** [comment]
If `mapSources` is a function currently it does [this](https://github.com/floridoo/gulp-sourcemaps/blob/master/src/write/index.internals.js#L40) . 

However, file is in scope so I could just pass the file or get the absolute path for you. If it is file then you can do whatever you want. Callback `(sourcePath, file) =>` .
 
Right [here](https://github.com/floridoo/gulp-sourcemaps/blob/master/src/write/index.internals.js#L57) absolute path is used and the cwd is then striped.

**Comment by nmccready** [comment]
Resolved: https://github.com/floridoo/gulp-sourcemaps/commit/40249b43376ddbfc067d6ef1d1d532f851052c84

Released in  1.11.1 and 2.4.1
